compat/src/index.d.ts

```diff
@@ -57,13 +58,13 @@ declare namespace React {
 	export function isValidElement(element: any): boolean;
 	export function findDOMNode(component: preact.Component): Element | null;
 
-	export class PureComponent<P = {}, S = {}> extends preact.Component {
+	export abstract class PureComponent<P = {}, S = {}> extends preact.Component {
 		isPureReactComponenet: boolean;
 	}
 
 	export function memo<P = {}>(component: preact.FunctionalComponent<P>, comparer?: (prev: P, next: P) => boolean): preact.FunctionComponent<P>;
 
-	export function forwardRef<P = {}>(fn: ForwardFn<P, any>): preact.FunctionalComponent<P>;
+	export function forwardRef<P = {}>(fn: _internal.ForwardFn<P, any>): preact.FunctionalComponent<P>;
```

Comment:
We offer forwarding a ref but we can't type the ref being forwarded and this seems to be possible in React under the following:

function forwardRef<T, P = {}>(
  Component: RefForwardingComponent<T, P>
): ComponentType<P & ClassAttributes<T>>